doc/sphinx/language/core/basic.rst

```diff
@@ -189,11 +189,18 @@ Other tokens
   tokens.
 
   When multiple tokens match the beginning of a sequence of characters,
-  the longest matching token is used.
+  the longest matching token not cutting a subsequence of contiguous letters in the middle is used.
   Occasionally you may need to insert spaces to separate tokens.  For example,
   if ``~`` and ``~~`` are both defined as tokens, the inputs ``~ ~`` and
-  ``~~`` generate different tokens, whereas if `~~` is not defined, then the
-  two inputs are equivalent.
+  ``~~`` generate different tokens, whereas if ``~~`` is not defined, then the
+  two inputs are equivalent. Also, if ``~`` and ``~_h`` are both
+  defined as tokens, the input ``~_ho`` is interpreted as ``~ _ho``
+  rather than ``~_h o`` so as not to cut the identifier-like
+  subsequence ``ho``. Contrastingly, if only ``~_h`` is defined as a token,
+  then ``~_ho`` is an error because no token can be found that includes
+  the whole subsequence ``ho`` without cutting it in the middle. Finally, if
+  all of ``~``, ``~_h`` and ``~_ho`` are defined as tokens, the input
+  ``~_ho`` is interpreted using the longest match rule, i.e. as the token ``~_ho``.
```

Comment:
I wanted to test this
```Coq
Set Printing All.

(* if ``~`` and ``~~`` are both defined as tokens,                                                                                                                                        
   the inputs ``~ ~`` and ``~~`` generate different tokens *)
Section TestLexer0.

Local Notation "~" := not.
Local Notation "~~" := not.

Check ~ ~ ⊤.
Check ~~ ⊤.

End TestLexer0.

(* whereas if ``~~`` is not defined,                                                                                                                                                      
   then the two inputs are equivalent *)
Section TestLexer1.

Local Notation "~" := not.

Set Printing All.
Check ~ ~ ⊤.
Check ~~ ⊤.

End TestLexer1.

(* Also, if ``~`` and ``~_h`` are both defined as tokens, the input                                                                                                                       
   ``~_ho`` is interpreted as ``~ _ho`` rather than ``~_h o`` so as                                                                                                                       
   not to cut the identifier-like subsequence ``ho``. *)
Section TestLexer2.

Local Notation "~" := not.
Local Notation "~_h" := (λ x ⇒ not (not x)).
Local Notation "'_ho'" := ⊥.
Let o := ⊤.

Check ~_ho.
Check ~_h o.

End TestLexer2.

(* Contrastingly, if only ``~_h`` is defined as a token, then ``~_ho``                                                                                                                    
   is an error because no token can be found that includes the whole                                                                                                                      
   subsequence ``ho`` without cutting it in the middle. *)
Section TestLexer3.

Local Notation "~_h" := (λ x ⇒ not (not x)).

Fail Check ~_ho.

End TestLexer3.

(* Finally, if all of ``~``, ``~_h`` and ``~_ho`` are defined as                                                                                                                          
   tokens, the input ``~_ho`` is interpreted using the longest match                                                                                                                      
   rule, i.e. as the token ``~_ho``. *)
Section TestLexer4.

Local Notation "~" := not.
Local Notation "~_h" := (λ x ⇒ not (not x)).
Local Notation "'_ho'" := ⊥.
Local Notation "~_ho'" := ⊤.

Check ~_ho.  (* gives "not False" instead of the expected "True" *)

End TestLexer4.
```
but the last test seems to contradict the last sentence?

Reply:
My bad, I did a stupid typo :(